Add Client login.html

harshnarang8 opened the pull request:
Summary:
Displays the login form when a user is redirected to /login.

Note:
Element id for Email is : inputEmail
Element id for Password is : inputPassword
Element id for Remember Me checkbox : inputCheckBox
The action of the form is /login i.e. the page itself.

Pool:

```
.______          ___            __  .______    __    __  .___________.    _______.
|   _  \        /   \          |  | |   _  \  |  |  |  | |           |   /       |
|  |_)  |      /  ^  \         |  | |  |_)  | |  |  |  | `---|  |----`  |   (----`
|      /      /  /_\  \  .--.  |  | |   ___/  |  |  |  |     |  |        \   \    
|  |\  \----./  _____  \ |  `--'  | |  |      |  `--'  |     |  |    .----)   |   
| _| `._____/__/     \__\ \______/  | _|       \______/      |__|    |_______/    
```

Commit:
Added iitk server login

Summary:
Uses ftp to verify user with iitk servers. Only the backend part.

How to test:
The complete login can be tested manually using `httpie`.
Run `node app` and then test using the following commands:
```
http get "localhost:8000"
http get "localhost:8000/profile"
```
^ Will redirect to `/login`

```
http post "localhost:8000/login" email="gangadhar@iitk.ac.in" password="shaktimaan"
```
^ If password is correct will redirect to `/profile` in the output

```
http post "localhost:8000/login" email="gangadhar@iitk.ac.in" password="batman"
```
^ If password is wrong will redirect to `/login`

Pool:
```
 __  __                 _   _
|  \/  | __ _ _ __ __ _| |_| |__   __ _ ___
| |\/| |/ _` | '__/ _` | __| '_ \ / _` / __|
| |  | | (_| | | | (_| | |_| | | | (_| \__ \
|_|  |_|\__,_|_|  \__,_|\__|_| |_|\__,_|___/
```

harshnarang8 committed:
Add basic login.html page without form

harshnarang8 committed:
Add basic login.html page without form

harshnarang8 committed:
Add form to login.html

harshnarang8 committed:
Add form to login.html

harshnarang8 committed:
Merge branch 'master' of https://github.com/harshnarang8/TechTests
for the update from main branch

harshnarang8 committed:
add footer login.html add user.html basic

harshnarang8 committed:
add footer login.html add user.html basic

harshnarang8 committed:
update login form with error message

harshnarang8 committed:
update login form with error message

harshnarang8 committed:
update login.html page layout

harshnarang8 committed:
update login.html page layout

harshnarang8 committed:
remove user.html empty file

harshnarang8 committed:
remove user.html empty file

harshnarang8 committed:
add id for inputCheckBox

harshnarang8 committed:
add id for inputCheckBox

pallavagarwal07 commented:
Please squash your commits into one if you are sure of the pull request

sakshamsharma commented:
Too much of clutter in the git commits. You are not supposed to keep more than the useful information in your commit messages. Would be helpful if you learnt to use `git commit --amend`, `git rebase` and `git rebase -i`.
@pallavagarwal07 I don't think it matters if he squashes them into 1. We can squash while merging as well. What matters is that he should change this whole thing into 2-3 commits meaningful or less if possible.

milindl committed:
Add frontend for login using ejs

Summary:
Finished yashsriv's backend with an ejs based front-end (login page and a test profile page for a successful login).

How to Test:
In the Server directory:
```
node app
```

Then direct your browser to localhost:8000 and test it out.
If password is correct, you're redirected to a test-profile page listing your access rights and username.

If your input is incorrect, you will be directed again to the /login page, with a toast displaying what went wrong on the top right.

Reason for Using EJS:
I wanted a decent way to manage views as the content would change with the user.
This seemed to be a quick and simple way to do that. Also, it looks quite neat.

Pool:
```
 __  __                 _   _
|  \/  | __ _ _ __ __ _| |_| |__   __ _ ___
| |\/| |/ _` | '__/ _` | __| '_ \ / _` / __|
| |  | | (_| | | | (_| | |_| | | | (_| \__ \
|_|  |_|\__,_|_|  \__,_|\__|_| |_|\__,_|___/

```

Commit:
Revert "Add frontend for login using ejs" since it is redundant

This reverts commit b043984d04db2ae0f3756e511b30a30128578541.
It is redundant as @harshnarang8 already has contributed something.

pallavagarwal07 committed:
Update README.md

harshnarang8 committed:
add id for inputCheckBox

harshnarang8 committed:
Merge branch 'master' of https://github.com/harshnarang8/TechTests

harshnarang8 committed:
merge login page commits

harshnarang8 committed:
Merge branch 'master' of https://github.com/harshnarang8/TechTests

pallavagarwal07 committed:
Merge pull request #2 from yashsriv/master

Add iitk server login

harshnarang8 committed:
Merge branch 'master' of https://github.com/harshnarang8/TechTests

kunalkap commented:
LGTM :+1:
For this pull request, Rajputs get 40 points, while 10 points go to Marathas for helping out.
